docs/lang/articles/contribution/dev_install.md

```diff
@@ -4,318 +4,467 @@ sidebar_position: 2
-:::
+<Tabs
+  defaultValue="ubuntu"
+  values={[
+    {label: 'macOS', value: 'macos'},
+    {label: 'Windows', value: 'windows'},
+    {label: 'Ubuntu', value: 'ubuntu'},
+    {label: 'Arch Linux', value: 'arch'},
+    {label: 'Other Linux distributions', value: 'others'},
+  ]}>
 
-   - If the previous LLVM binaries do not work, please build from source:
-     - For Linux & Mac OSX:
-
-       ```bash
-       wget https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/llvm-10.0.0.src.tar.xz
-       tar xvJf llvm-10.0.0.src.tar.xz
-       cd llvm-10.0.0.src
-       mkdir build
-       cd build
-       cmake .. -DLLVM_ENABLE_RTTI:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=OFF -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86;NVPTX" -DLLVM_ENABLE_ASSERTIONS=ON -DLLVM_ENABLE_TERMINFO=OFF
-       # If you are building on Apple M1, use -DLLVM_TARGETS_TO_BUILD="AArch64".
-       # If you are building on NVIDIA Jetson TX2, use -DLLVM_TARGETS_TO_BUILD="ARM;NVPTX"
-       # If you are building for a PyPI release, add -DLLVM_ENABLE_Z3_SOLVER=OFF to reduce the library dependency.
-       make -j 8
-       sudo make install
-       # Check your LLVM installation
-       llvm-config --version  # You should get 10.0.0
-       ```
-
-     - For Windows:
-
-       ```bash
-       # For Windows
-       # LLVM 10.0.0 + MSVC 2019
-       cmake .. -G "Visual Studio 16 2019" -A x64 -DLLVM_ENABLE_RTTI:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=OFF   -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86;NVPTX" -DLLVM_ENABLE_ASSERTIONS=ON -Thost=x64   -DLLVM_BUILD_TESTS:BOOL=OFF -DCMAKE_INSTALL_PREFIX=installed
-       ```
-
-       - Then open `LLVM.sln` and use Visual Studio 2017+ to build.
-       - Please make sure you are using the `Release` configuration.
-         After building the `INSTALL` project (under folder
-         `CMakePredefinedTargets` in the Solution Explorer window).
-       - If you use MSVC 2019, **make sure you use C++17** for the
-         `INSTALL` project.
-       - After the build is complete, find your LLVM binaries and
-         headers in `build/installed`.
-       - Please add `build/installed/bin` to `PATH`. Later, when you build Taichi, please use `cmake -DLLVM_DIR=<path_to_build>/build/installed/lib/cmake/llvm`.
-
-
-### Setting up CUDA (optional)
+<TabItem value="macos">
+
+1. Ensure that the Clang that ships with your MacBook has a version &ge;8 and &lt;12:
+
+  ```
+  clang --version
+  ```
+
+2. If your Clang version is &ge;12, install Clang 11:
+
+  ```
+  brew install llvm@11
+  export CXX=/opt/homebrew/opt/llvm@11/bin/clang++
+  ```
+
+</TabItem>
+
+<TabItem value="windows">
+
+Skip this section because we provide scripts that will install `Clang` for you. See the [Build Taichi from source](#build-taichi-from-source) for more information. 
+
+</TabItem>
+
+<TabItem value="ubuntu">
+
+```
+sudo apt install clang-10
```

Comment:
Is this included in every version of Ubuntu distribution? IIRC, users may need to manually add the https://apt.llvm.org repository.

Reply:
@ailzhang Could you double check?

Reply:
let's add an item in debugging section, sth like: 
if your `sudo apt install clang-10` failed, consider adding the apt source following https://apt.llvm.org/